src/main/java/ru/curs/lyra/service/DataFactory.java

```diff
@@ -89,17 +89,13 @@ private void setLyraExactTotalCount() {
 
         if (lyraApproxTotalCountBeforeGetRows < basicGridForm.getGridHeight() + LYRA_EXACT_TOTALCOUNT_DELTA) {
 
-            if (lyraGridAddInfo.getDgridOldTotalCount() > 0) {
-                lyraApproxTotalCountBeforeGetRows = lyraGridAddInfo.getDgridOldTotalCount();
-            } else {
-                final int[] array = new int[1];
-                basicGridForm.externalAction(c -> {
-                    array[0] = c.count();
-                    return null;
-                }, null);
+            final int[] array = new int[1];
```

Comment:
if you want to mutate a single value outside lambda it's better to use `AtomicInteger`

All this `externalAction` thing will soon be removed, though.